src/Symfony/Component/Serializer/Tests/Normalizer/AbstractObjectNormalizerTest.php

```diff
@@ -40,6 +42,23 @@ public function testInstantiateObjectDenormalizer()
         $this->assertInstanceOf(__NAMESPACE__.'\Dummy', $normalizer->instantiateObject($data, $class, $context, new \ReflectionClass($class), array()));
     }
 
+    public function testDenormalizeWithDiscriminatorAndPropertyInfo()
+    {
+        $propertyTypeExtractor = $this->createMock(PropertyTypeExtractorInterface::class);
+        $propertyTypeExtractor->expects($this->exactly(2))
+            ->method('getTypes')
+            ->withConsecutive(
+                array(MappedDummyChild::class, 'foo'),
+                array(MappedDummyChild::class, 'bar')
+            );
+
+        $normalizer = new AbstractObjectNormalizerWithMetadata($propertyTypeExtractor);
+
+        $data = array('foo' => 'dummy', 'bar' => 1);
```

Comment:
Can you not create these variables and directly call `denormalize` with the values?